content/blog/ggplot2-styling/index.Rmd

```diff
@@ -0,0 +1,1141 @@
+    box.background = element_rect(fill = "cornsilk"),
+    box.margin = margin_auto(5),
+    
+    # Tweaking the justification per position
+    justification.left = "top",
+    justification.right = "bottom"
+  )
+```
+
+#### General legend guides
+
+Moving on from guide boxes to the guides themselves;
+There are some theme settings that (almost) universally affect any guides, regardless of `guide_legend()`, `guide_colourbar()`, or `guide_bins()`.
+These settings pertain to the legend background, margins, labels and titles and their placement and key sizes.
+
+```{r sub_legend_general}
+p +
+  theme_sub_legend(
+    # Give guides a wider background
+    background = element_rect(fill = "cornsilk"),
+    margin = margin_auto(5, unit = "mm"),
+    
+    # Display legend titles to the right of the guide
+    title = element_text(angle = 270),
+    title.position = "right",
+    
+    # Display red labels to the left of the keys
+    text = element_text(colour = "red"),
+    text.position = "left",
+    
+    # Set smaller keys
+    key.width = unit(5, "mm"),
+    key.height = unit(5, "mm")
+  )
+```
+
+#### Legend guide
+
+There are also settings that affect `guide_legend()` but not `guide_colourbar()`.
+Most of these have to do with the arrangement of keys, like their spacing, justification or fill order (by row or column).
+The `legend.key.justification` setting only matters when the text size exceeds the key size.
+If we remove that setting from the plot below, the keys will fill up to fit the space.
+
+```{r sub_legend_legend}
+p + 
+  # Set two columns and long label text
+  scale_shape_discrete(
+    labels = c("4\nwheel\ndrive", "front\nwheel\ndrive", "rear\nwheel\ndrive"),
+    guide = guide_legend(ncol = 2)
+  ) +
+  theme_sub_legend(
+    # Fill items in grid in a row-wise fashion
+    byrow = TRUE,
+    # Increase spacing between keys
+    key.spacing.y = unit(5, "mm"),
+    key.spacing.x = unit(5, "mm"),
+    # Top-align keys with text
+    key.justification = "top"
+  )
+```
+
+#### Colourbar guide
+
+Likewise, there are also settings specific to `guide_colourbar()`.
+Generally, you can see it as a legend guide with a single elongated key.
+This elongation has special behaviour in that the default is 5 times the original key size.
+If you need to set the size directly without special behaviour, you can use the `guide_colourbar(theme)` argument.
+Aside from the special size behaviour, we can also set the colourbar frame and ticks.
+
+```{r sub_legend_colourbar}
+p +
+  # Using a local guide theme to directly set the size
+  guides(colour = guide_colourbar(theme = theme(legend.key.height = unit(5, "cm")))) +
+  theme_sub_legend(
+    frame = element_rect(colour = "red"),
+    # Long blue ticks
+    ticks = element_line(colour = "blue"),
+    ticks.length = unit(-5, "mm"),
+    # Adapt margins to accommodate longer ticks
+    text = element_text(margin = margin(l = 6, unit = "mm")),
+    margin = margin(l = 6, unit = "mm")
+  )
+```
```

Comment:
Maybe mention here how you can make the bar stretch the full height/width of the plot